### ISSUE TITLE
The problem with starting Rust server

### ISSUE THREAD

**Issue by xMasterX** [dgibbs64/linuxgsm#835]
Debian 8 - 64bit - 1.5gb ram

After installing dependencies, and installing server using (./rustserver install)

I tried to run server :

> rustserver@debian-8-64bit:~$ ./rustserver st
> [ INFO ] Log Manager rust-server: Moving game logs to /home/rustserver/log/server
> [  OK  ] Starting rust-server: TestRust

But once the server is turned off and i got this log
[gamelog.txt](https://github.com/dgibbs64/linuxgsm/files/282217/gamelog.txt)

Debug don't working :

> [  OK  ] Debug rust-server: Starting debug
> Set current directory to /home/rustserver/serverfiles
> Found path: /home/rustserver/serverfiles/RustDedicated
> /home/rustserver/lgsm/functions/command_debug.sh: line 59: 24486 Aborted                 ${executable} ${parms}

What do you advise? It is necessary to increase the memory up to 4~8 GB?

**Comment by UltimateByte** [comment]
Rust takes up several GB, even with small maps. So lack of RAM might be one of the reasons. 
SIGSEGV errors can be anything that makes mono runtime crash, including lack of RAM.

**Comment by UltimateByte** [comment]
`/home/rustserver/lgsm/functions/command_debug.sh: line 59: 24486 Aborted ${executable} ${parms}`
=> This is not normal. Update your LGSM functions.
`./rustserver uf`

**Comment by xMasterX** [comment]
I hope this problem caused by RAM.
After update debug still not working

> [  OK  ] Debug rust-server: Starting debug
> Set current directory to /home/rustserver/serverfiles
> Found path: /home/rustserver/serverfiles/RustDedicated
> /home/rustserver/lgsm/functions/command_debug.sh: line 59: 25326 Aborted                 ${executable} ${parms}

**Comment by UltimateByte** [comment]
I'l using debian 8.4 too, but never with less than 32GB RAM, so i couldn't tell if it's the reason. I've seen many errors, but never that one. 1.5GB RAM is too few anyways.

**Comment by PuggerHugger** [comment]
Also using debian 8.4, server says its starting up but i can never connect to the server through the ip.... I am positive i did everything correctly.

**Comment by UltimateByte** [comment]
@PuggerHugger Server specs ? Distro ? Logs ?

**Comment by ravermeister** [comment]
Hi, I'm having similar problems:
(dependencies are installed, and the new tmux version too)
Server is: Ubuntu 14.04 lts 64bit

`gaming@echo390:~/testing/rust$ ./rustserver validate
    fetching command_validate.sh...OK
[ WARN ] Validate rust-server: Validating may overwrite some customised files.
https://developer.valvesoftware.com/wiki/SteamCMD#Validate
[  OK  ] Validate rust-server: Checking server filesERROR: ld.so: object '/usr/lib/coreutils/libstdbuf.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
Redirecting stderr to '/opt/gaming/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...Created shared memory when not owner SteamController_Shared_mem
OK.

Connecting anonymously to Steam Public...Logged in OK
Waiting for license info...OK
 Update state (0x5) validating, progress: 0,00 (5365 / 3130516841)
 Update state (0x5) validating, progress: 2,62 (82036969 / 3130516841)
 Update state (0x5) validating, progress: 6,81 (213182232 / 3130516841)
 Update state (0x5) validating, progress: 12,96 (405689668 / 3130516841)
 Update state (0x5) validating, progress: 19,10 (598034534 / 3130516841)
 Update state (0x5) validating, progress: 27,39 (857557094 / 3130516841)
 Update state (0x5) validating, progress: 35,38 (1107642470 / 3130516841)
 Update state (0x5) validating, progress: 38,31 (1199392870 / 3130516841)
 Update state (0x5) validating, progress: 41,41 (1296386150 / 3130516841)
 Update state (0x5) validating, progress: 47,87 (1498720047 / 3130516841)
 Update state (0x5) validating, progress: 51,89 (1624319840 / 3130516841)
 Update state (0x5) validating, progress: 55,71 (1743857504 / 3130516841)
 Update state (0x5) validating, progress: 59,56 (1864443744 / 3130516841)
 Update state (0x5) validating, progress: 64,30 (2012847603 / 3130516841)
 Update state (0x5) validating, progress: 69,31 (2169609715 / 3130516841)
 Update state (0x5) validating, progress: 73,81 (2310643187 / 3130516841)
 Update state (0x5) validating, progress: 78,32 (2451676659 / 3130516841)
 Update state (0x5) validating, progress: 81,80 (2560728563 / 3130516841)
 Update state (0x5) validating, progress: 85,15 (2665586163 / 3130516841)
 Update state (0x5) validating, progress: 89,30 (2795609587 / 3130516841)
 Update state (0x5) validating, progress: 93,22 (2918292979 / 3130516841)
 Update state (0x5) validating, progress: 96,49 (3020529139 / 3130516841)
 Update state (0x5) validating, progress: 98,58 (3086050296 / 3130516841)
Success! App '258550' fully installed.`


but when I try to Start I get this:

`Success! App '258550' fully installed.
gaming@echo390:~/testing/rust$ ./rustserver debug
    fetching command_debug.sh...OK

Rust Debug
=================================

Distro: Ubuntu 14.04.4 LTS
Arch: x86_64
Kernel: 3.13.0-88-lowlatency
Hostname: echo390

Start parameters:
./RustDedicated -batchmode +server.ip 123.123.123 +server.port 5555 +server.tickrate 30 +server.hostname "Rust" +server.identity "rust-server"  +server.maxplayers 50 +server.worldsize 4000 +server.saveinterval 300 +rcon.ip 123.123.123 +rcon.port 44444 +rcon.password "CHANGE_ME" -logfile "gamelog-2016-06-26-15-57-09.log"

Use for identifying server issues only!
Press CTRL+c to drop out of debug mode.
Warning! If rust-server is already running it will be stopped.

Continue? [Y/n]y
[ INFO ] Debug rust-server: Stopping any running servers
    fetching command_stop.sh...OK
[  OK  ] Stopping rust-server: Rust is already stopped
[  OK  ] Debug rust-server: Starting debug
Set current directory to /opt/gaming/testing/rust/serverfiles
Found path: /opt/gaming/testing/rust/serverfiles/RustDedicated`

and then nothing. the Server is simply not running. What am I doing wrong?

**Comment by dgibbs64** [comment]
The server is running. Thats all the console outputs for that type of game engine. Unlike source engine that gives you lots of feedback

**Comment by ravermeister** [comment]
But I neither can Connect to the Server nor do I See the RustDedicated Server process

**Comment by ravermeister** [comment]
and starting the Server normally gives me this:

```
gaming@gameserver:~/server/rust$ server.control start
[ INFO ] Log Manager rust-server: Moving game logs to /opt/gaming/server/85.25.138.90_rust_32110/log/server
[  OK  ] Update rust-server: Checking for update: SteamCMD

No update available:
        Current version: 1195523
        Available version: 1195523
        https://steamdb.info/app/258550/

[  OK  ] Update rust-server: No update available
[ FAIL ] Starting rust-server: Unable to start Rust Dedicated Server
```

**Comment by UltimateByte** [comment]
1) To few info and no logs (console is empty in Rust, but logs aren't)
2) Your issue is not similar
3) General support is here https://steamcommunity.com/groups/linuxgsm/discussions

Open a topic there and give relevant info, masking your Rcon password, and outputing interesting stuff to pastebin, like logs and the output of ./rustserver details
